pyquil/quilbase.py

```diff
@@ -199,11 +199,23 @@ def out(self) -> str:
                 _format_qubits_out(self.qubits),
             )
 
-    def controlled(self, control_qubit: QubitDesignator) -> "Gate":
+    def controlled(self, control_qubits: Union[QubitDesignator, List[QubitDesignator]]) -> "Gate":
```

Comment:
```suggestion
    def controlled(self, control_qubits: Union[QubitDesignator, Sequence[QubitDesignator]]) -> "Gate":
```

Prefer `Sequence` to `List` for typing input parameters.

https://docs.python.org/3/library/typing.html#typing.List

If you switch from recursion to iteration below, you could even broaden the type to `Iterable`

Reply:
I'll try to fix it to use Iterable.

Reply:
Off-topic: I was actually surprise how pyquil is demanding of correct types. It looks so much like trying to achieve Julia's type hierarchy. Especially taking into account the duck-typing convention in python.

Reply:
There has been a recent push to add type hints everywhere. There are pros and cons, but on the whole it helps to document the types and prevent certain classes of bugs.

Reply:
Done